mutiny-core/src/nodemanager.rs

```diff
@@ -933,19 +953,25 @@ impl NodeManager {
     }
 
     pub async fn list_peers(&self) -> Result<Vec<MutinyPeer>, MutinyError> {
-        let nodes = self.nodes.lock().await;
+        let peer_data = gossip::get_all_peers().await?;
 
         // get peers saved in storage
-        let mut storage_peers: Vec<MutinyPeer> = nodes
+        let mut storage_peers: Vec<MutinyPeer> = peer_data
             .iter()
-            .flat_map(|(_, n)| n.persister.list_peer_connection_info())
-            .map(|(pubkey, connection_string)| MutinyPeer {
-                pubkey,
-                connection_string,
+            .map(|(node_id, metadata)| MutinyPeer {
+                // node id should be safe here
+                pubkey: secp256k1::PublicKey::from_slice(node_id.as_slice())
+                    .expect("Invalid pubkey"),
+                connection_string: metadata.connection_string.clone(),
+                alias: metadata.alias.clone(),
+                color: metadata.color.clone(),
+                label: metadata.label.clone(),
                 is_connected: false,
             })
             .collect();
```

Comment:
Wait this could just be a side effect with a bug (or not intended behavior) with our existing API. We're just doing a normal `list_peers`, but perhaps we should change the API to indicate a specific node and the peers that node should be connected to?